ssl/ssl_local.h

```diff
@@ -603,6 +603,7 @@ struct ssl_session_st {
     CRYPTO_REF_COUNT references;
     OSSL_TIME timeout;
     OSSL_TIME time;
+    OSSL_TIME rtt;
```

Comment:
I do not think it should be part of the SSL_SESSION object.

Reply:
Thanks for pointing this out. Where would you recommend the TLS RTT value be if not in the session object?

Reply:
I am afraid there is no way. The SSL_SESSION objects should be immutable once created as they can be used by multiple threads when resuming sessions.

Reply:
Maybe @mattcaswell has some opinion on this?

Reply:
> If there's already a method for retrieving the most recent ssl object of a session, we haven't quite found it.

There isn't. You *can* associate arbitrary data to a session via the `SSL_SESSION_set_ex_data()` function (or the `SSL_SESSION_set_app_data` macro which uses `SSL_SESSION_set_ex_data()` underneath) which can later be retrieved via `SSL_SESSION_get_ex_data` (or `SSL_SESSION_get_app_data`). So plausibly Nginx could store the RTT value in the SSL_SESSION itself via this route before passing it to the TLS logging module - but that throws up questions about locking since the SSL_SESSION is likely to be in the session cache and therefore could be shared by multiple threads. It also begs the question about what happens if a single SSL_SESSION is shared by multiple connections at the same time (which may have different calculated RTT values)

Anyway, @t8m is right, the correct location for this to be stored is the ssl_connection.

Reply:
We agree that putting the RTT in the SSL object is the logical answer since this is a per-connection metric. We'll do that and associate the connections to a session on the web server end instead.. In retrospect, we see that it wasn't smart to try to associate connections with a session in OpenSSL when the webserver should be handling that. 

Also, an edit to what I said earlier, NGINX does not always pass in sessions, it works with SSL objects in the logging much more frequently, so I was completely wrong there when I said that it passes in sessions often. 

This might be a trivial problem, but how can the `ssl_connection` be accessed using an `ssl` object? Or should we go the route of placing the RTT directly in the `ssl` object seeing as Nginx uses these?

Reply:
> how can the ssl_connection be accessed using an ssl object?

Just use the `SSL_CONNECTION_FROM_SSL` macro to get the SSL_CONNECTION object.

> Or should we go the route of placing the RTT directly in the ssl object seeing as Nginx uses these?

It's TLS specific (as opposed to QUIC) so it belongs in the `SSL_CONNECTION`.

Reply:
> I do not think it should be part of the SSL_SESSION object.

Well web3nfullstack prerepozitories fór pull requests updial/uphold dial verbosing

Reply:
Perfect, we've done it and it works well with our NGINX implementation! Thanks for correcting our glaring mistake.

Should I commit my changes to my master branch and could you can see it if I did? Or should I scrap this pull request and make a new one? I apologize for my unfamiliarity with pull requests.

Reply:
You can reuse this PR or create a new one. I'd suggest though to use a different branch than master which implies new PR. Also please never add merge commits. Instead use git rebase and force push if you need to rebase against the latest master branch to resolve conflicts.